app/models/jira_board.rb

```diff
@@ -28,7 +29,26 @@ class JiraBoard < ApplicationRecord
 
   has_many :jira_issues, dependent: :destroy
   has_many :jira_sprints, dependent: :destroy
+  has_many :jira_environments, dependent: :destroy
 
-  validates :jira_project_key, presence: true, uniqueness: true, allow_blank: true
+  validates :jira_project_key, presence: true, allow_blank: true
   validates :jira_board_id, uniqueness: true, allow_blank: true
+
+  accepts_nested_attributes_for :jira_environments
+
+  def jira_environments_development
+    jira_environments.where(environment: 'development').map(&:custom_environment)
+  end
+
+  def jira_environments_staging
+    jira_environments.where(environment: 'staging').map(&:custom_environment)
+  end
+
+  def jira_environments_qa
+    jira_environments.where(environment: 'qa').map(&:custom_environment)
+  end
+
+  def jira_environments_production
+    jira_environments.where(environment: 'production').map(&:custom_environment)
+  end
```

Comment:
1. Can we avoid using `map` and use a SQL function like `select` here? Just asking
2. I see duplicated code here, maybe creating a method that receives the environment param?
3. Is there a chance to move this to a query object?